core/services/pipeline/task_params.go

```diff
@@ -717,7 +717,15 @@ func (s *AddressSliceParam) UnmarshalPipelineParam(val interface{}) error {
 
 type JSONPathParam []string
 
+// UnmarshalPipelineParam unmarshals a slice of strings from val.
+// If val is a string or []byte, it is split on a separator.
+// The default separator is ',' but can be overridden by initializing p with a single element before calling.
 func (p *JSONPathParam) UnmarshalPipelineParam(val interface{}) error {
```

Comment:
I do agree we need a test here, but I can add tests during my ongoing "refactoring" later.